### DISCUSSION TITLE
Easily editing wim file

### DISCUSSION THREAD

**Discussion by Tarek-Hasan** [CodingWonders/DISMTools#95] [General]
I use imagex tool in Dism++ to edit and remove unnecessary edition in windows iso. Just have to load up the wim and edit it.

![Screenshot 2024-01-09 122102](https://github.com/CodingWonders/DISMTools/assets/27731012/b2180044-e71c-4024-90bb-8da84957b2ee)

When I found your open source DISM app, I looked for a similar workflow as mine in it. But can't seem to find it.
Is this type of feature exist in this tool?
If yes, please help me. If no, is there a chance you may add this feature?

**Comment by CodingWonders** [comment]
I haven't used DISM++, so I don't know what you mean about editing an image. Maybe you're talking about editing the XML metadata? Can you show this functionality?

**Comment by Tarek-Hasan** [comment]
Microsoft ships windows ISOs in multi-edition pack meaning it contains 10 different editions like education, core, pro etc. I like to keep only pro edition in the ISO I use. So I get rid of all unnecessary edition by loading the WIM file in DISM++ and exporting the desired edition. There's also a remove option, as you can see on the screenshot, but I don't use it. As removing or deleting a file from WIM doesn't actually remove it, this just makes the file an unindexed garbage.
Sorry, I currently don't have any multi-edition ISO to show you the process. Here, is the link for [DISM++](https://github.com/Chuyu-Team/Dism-Multi-language/releases) if you feel curious to try it out.

Note: DISM++ is not open source. It uses github for software distribution only.

**Comment by CodingWonders** [comment]
> Microsoft ships windows ISOs in multi-edition pack meaning it contains 10 different editions like education, core, pro etc. I like to keep only pro edition in the ISO I use. So I get rid of all unnecessary edition by loading the WIM file in DISM++ and exporting the desired edition. There's also a remove option, as you can see on the screenshot, but I don't use it. As removing or deleting a file from WIM doesn't actually remove it, this just makes the file an unindexed garbage.
> Sorry, I currently don't have any multi-edition ISO to show you the process. Here, is the link for [DISM++](https://github.com/Chuyu-Team/Dism-Multi-language/releases) if you feel curious to try it out.
> 
> Note: DISM++ is not open source. It uses github for software distribution only.

Yes, I know that because the program does support removing unnecessary editions. But how do you edit the XML metadata with DISM++?

**Comment by Tarek-Hasan** [comment]
To my knowledge, wimgapi.dll and wofadk.sys libraries included in WADK tools which are used by DISM++ and MSMG ToolKit does that for you. There's also an open source alternative for Windows Imaging if you are interested, [wimlib](https://wimlib.net/).

**Comment by CodingWonders** [comment]
I took the liberty of converting this discussion to an issue for better progress tracking (#96). We'll discuss things there from here on.
